Add troubleshooting guide for upgrade (#7396)
wazuh/wazuh-documentation: source/upgrade-guide/troubleshooting.rst

.. Copyright (C) 2015, Wazuh, Inc.

.. meta::

>   :description: This section collects common issues that may occur when upgrading central components, along with basic steps to solve them.

Reviewer: ```suggestion
   :description: This section collects common issues that might occur when upgrading central components and provides steps to resolve them.
```

> .. _central_components_troubleshooting:
> 

Reviewer: ```suggestion
```

Troubleshooting
===============

> This section collects common issues that may occur when upgrading central components, along with basic steps to solve them.

Reviewer: ```suggestion
This section collects common issues that might occur when upgrading central components and provides steps to resolve them.
```

No username and password found in the keystore
----------------------------------------------

For alerts and vulnerabilities detected by Vulnerability detection to be indexed and displayed on the dashboard, it is necessary to add the credentials of the indexer in the manager keystore. These credentials are those obtained at the time of installation.

.. code-block:: console

   # /var/ossec/bin/wazuh-keystore -f indexer -k username -v <INDEXER_USERNAME>
   # /var/ossec/bin/wazuh-keystore -f indexer -k password -v <INDEXER_PASSWORD>

IndexerConnector initialization failed
--------------------------------------

This warning may be due to incorrect keystore credentials, as mentioned in the previous section, or there may be a configuration or certificate error. To solve this it is necessary to check that the :doc:`indexer section</user-manual/reference/ossec-conf/indexer>` is correctly configured in ``/var/ossec/etc/ossec.conf`` by checking that the IP, port and certificates path are as expected.

> Once the error has been fixed and it has been possible to connect to the indexer, we will be able to see a log like the following one:

Reviewer: ```suggestion
After fixing the error and successfully connecting the Wazuh manager to the Wazuh indexer, you can see a log similar to the following:
```

> .. code-block:: console

Reviewer: ```suggestion
.. code-block:: none
```

   INFO: IndexerConnector initialized successfully for index: ...

If the error persists, temporarily enable ``wazuh_modules.debug=2`` in ``/var/ossec/etc/local_internal_options.conf`` to get more information on what may be causing the issue.

Vulnerability detection seems to be disabled or has a problem
-------------------------------------------------------------

> This warning shows that Vulnerability detection may be disabled or there is an error in the configuration. To check what may be happening, we must first check that ``vulnerability-detection`` is enabled in ``/var/ossec/etc/ossec.conf``. If it is enabled, we have to check that the vulnerability index ``wazuh-states-vulnerabilities-*`` has been created correctly. This can be checked in ``Indexer Management > Index Management > Indices``.
> 
> If the index has not been created, check the manager logs for any errors or warnings, as this may be due to an error mentioned in the previous sections:

Reviewer: ```suggestion
This warning indicates that the Vulnerability Detection module might be disabled or there could be a configuration error. To find out the issue, first ensure that ``vulnerability-detection`` is enabled in ``/var/ossec/etc/ossec.conf``. If it's enabled, verify that the vulnerability index ``wazuh-states-vulnerabilities-*`` has been correctly created. You can check this under **Indexer Management** > **Index Management** > **Indices** configuration.

If the index hasn't been created, check the Wazuh manager logs for any errors or warnings, as the issue might be related to errors mentioned in previous sections:
```

.. code-block:: console

    # cat /var/ossec/logs/ossec.log | grep -i -E "error|warn"

Application Not Found
---------------------

> If after updating, when accessing the dashboard we find the message ``Application Not Found``, this may be because after the update the file ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` has not been overwritten with the new changes. To fix this, please add the following line to the configuration file:

Reviewer 2: ```suggestion
If after updating, when accessing the dashboard we find the message ``Application Not Found``, it is possible that the ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` configuration file has not been overwritten with the new changes after the upgrade. To fix this, replace the following line in the configuration file:
```

Reviewer: ```suggestion
If you encounter the message *Application Not Found* when accessing the Wazuh dashboard after upgrading, it might be that the configuration file ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` wasn't overwritten with new changes. To resolve this issue, replace the following line in the configuration file:
```

> .. code-block:: console
> 
>     uiSettings.overrides.defaultRoute: /app/wz-home
> 

Reviewer: ```suggestion
.. code-block:: none

   uiSettings.overrides.defaultRoute: /app/wz-home
```

None of the above solutions are fixing my problem
-------------------------------------------------

We have a welcoming community that can help you with most of the problems you might have regarding Wazuh deployment and usage `<https://wazuh.com/community>`_.

Also, you can contact us for opening issues in our GitHub repositories under the `organization <https://github.com/wazuh>`_.

We will be interested in the log files of your deployment. You can check them out on each component:

Check the following log files:

 - Wazuh indexer:

      .. code-block:: console

          # cat /var/log/wazuh-indexer/wazuh-cluster.log | grep -i -E "error|warn"

 - Wazuh manager:

      .. code-block:: console

          # cat /var/log/filebeat/filebeat | grep -i -E "error|warn"

          # cat /var/ossec/logs/ossec.log | grep -i -E "error|warn"

 - Wazuh dashboard:

      .. code-block:: console

          # journalctl -u wazuh-dashboard

          # cat /usr/share/wazuh-dashboard/data/wazuh/logs/wazuhapp.log | grep -i -E "error|warn"

    .. note::
      The Wazuh indexer uses the ``/var/log`` folder to store logs by default.

    .. warning::

>       By default, the Wazuh dashboard doesn't store logs on a file. You can change this by configuring ``logging.dest`` setting in the ``opensearch_dashboard.yml`` configuration file.

Reviewer 2: I do not think that this should be an approach, we have more log files that may not be related to the problem, so we should suggest providing all related info that the user can share.

.. Copyright (C) 2015, Wazuh, Inc.

.. meta::
   :description: This section collects common issues that might occur when upgrading central components and provides steps to resolve them.

Troubleshooting
===============

This section collects common issues that might occur when upgrading central components and provides steps to resolve them.

The 'vulnerability-detector' configuration is deprecated
--------------------------------------------------------

> This warning appears because when upgrading the Wazuh manager, the ``/var/ossec/etc/ossec.conf`` file remains unchanged, retaining the previous configuration of the Vulnerability Detection module. In addition, invalid configuration warnings might appear for the ``interval``, ``min_full_scan_interval``, ``run_on_start`` and ``provider`` elements. To resolve this issue, update the configuration as specified in the :doc:`Vulnerability detection documentation </user-manual/capabilities/vulnerability-detection/configuring-scans>`.

Reviewer: ```suggestion
This warning appears because when upgrading the Wazuh manager, the ``/var/ossec/etc/ossec.conf`` file remains unchanged, retaining the previous configuration of the Vulnerability Detection module. In addition, invalid configuration warnings might appear for the ``interval``, ``min_full_scan_interval``, ``run_on_start`` and ``provider`` elements. To resolve this issue, update the configuration as specified in :doc:`/user-manual/capabilities/vulnerability-detection/configuring-scans`.
```

No username and password found in the keystore
----------------------------------------------

To ensure that alerts and vulnerabilities detected by the Vulnerability Detection module are indexed and displayed on the Wazuh dashboard, you need to add the credentials of the Wazuh indexer to the Wazuh manager keystore.

.. code-block:: console

   # /var/ossec/bin/wazuh-keystore -f indexer -k username -v <INDEXER_USERNAME>
   # /var/ossec/bin/wazuh-keystore -f indexer -k password -v <INDEXER_PASSWORD>

IndexerConnector initialization failed
--------------------------------------

This warning might be due to incorrect keystore credentials or indicate a configuration or certificate error. To resolve this, ensure that the IP address, port, and certificate paths are configured correctly in the :doc:`indexer section</user-manual/reference/ossec-conf/indexer>` in ``/var/ossec/etc/ossec.conf``.

After fixing the error and successfully connecting the Wazuh manager to the Wazuh indexer, you can see a log similar to the following:

.. code-block:: none

   INFO: IndexerConnector initialized successfully for index: ...

> If the error persists, temporarily enable ``wazuh_modules.debug=2`` in ``/var/ossec/etc/local_internal_options.conf`` to get more information on what may be causing the issue.

Reviewer: ```suggestion
To get more information if the error persists, temporarily enable ``wazuh_modules.debug=2`` in ``/var/ossec/etc/local_internal_options.conf``.
```

Vulnerability detection seems to be disabled or has a problem
-------------------------------------------------------------

> This warning indicates that the Vulnerability Detection module might be disabled or there could be a configuration error. To find out the issue, first ensure that ``vulnerability-detection`` is enabled in ``/var/ossec/etc/ossec.conf``. If it's enabled, verify that the vulnerability index ``wazuh-states-vulnerabilities-*`` has been correctly created. You can check this under **Indexer Management** > **Index Management** > **Indices** configuration.
> 
> If the index hasn't been created, check the Wazuh manager logs for any errors or warnings, as the issue might be related to errors mentioned in previous sections:
> 
> .. code-block:: console
> 
>    # cat /var/ossec/logs/ossec.log | grep -i -E "error|warn"
> 

Reviewer: ```suggestion
This warning indicates that the Vulnerability Detection module might be disabled or there could be a configuration error. To troubleshoot:

#. Ensure that ``vulnerability-detection`` is enabled in ``/var/ossec/etc/ossec.conf``.
#. Verify that the vulnerability index ``wazuh-states-vulnerabilities-*`` has been correctly created. You can check this under **Indexer Management** > **Index Management** > **Indices** configuration.
#. If the index wasn't created, check the Wazuh manager logs for any errors or warnings, as the issue might be related to errors mentioned in previous sections.

   .. code-block:: console
   
      # cat /var/ossec/logs/ossec.log | grep -i -E "error|warn"
```

Application Not Found
---------------------

> If you encounter the message *Application Not Found* when accessing the Wazuh dashboard after upgrading, it might be that the configuration file ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` wasn't overwritten with new changes. To resolve this issue, update the ``uiSettings.overrides.defaultRoute`` setting value with ``/app/wz-home`` in the configuration file:

Reviewer: ```suggestion
If you encounter the message *Application Not Found* when accessing the Wazuh dashboard after upgrading, it might be that the configuration file ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` wasn't overwritten with new changes. To resolve this issue, update the ``uiSettings.overrides.defaultRoute`` setting with the ``/app/wz-home`` value in the configuration file:
```
